indexing-hadoop/src/main/java/org/apache/druid/indexer/InputRowSerde.java

```diff
@@ -190,7 +215,7 @@ public void serialize(ByteArrayDataOutput out, Object value)
     @Override
     public Long deserialize(ByteArrayDataInput in)
     {
-      return in.readLong();
+      return isNullByteSet(in) ? null : in.readLong();
```

Comment:
Perhaps it would be better to use a functional programming style here.

```java
return Optional.ofNullable(in)
               .filter(InputRowSerde::isNullByteSet)
               .map(ByteArrayDataInput::readLong)
               .get();
```